src/jit/morph.cpp

```diff
@@ -12829,6 +12829,26 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
             break;
 #endif
 
+        case GT_ARR_LENGTH:
+            // Morph "constant_string".Length to CNS_INT(15)
+            if (op1->OperIs(GT_IND) && fgGlobalMorph)
```

Comment:
Does this really have to wait until morph? Is there any way to do this during importation?

Reply:
I wonder if this is enough to catch all cases. Does this work if the string is a `static readonly` field? Does it work if you have conditional code like `(c ? "abc" : "def").Length`? The later, while probably not common, would likely require updating VN to work.

Reply:
@mikedn if I do it in the importation phase will it handle e.g. `stringBuilder.Append(".");` ? (Append will be inlined and it checks for Length < 2 inside afair for fast insert -- so with the current change it's improved). However, I suspect I need to save string length to CNS_STR there.. if it's possible

Reply:
The inliner will directly substitute constants like this, so checking in the importer should get most cases.